### ISSUE TITLE
Suptitle positioning messed up in 3.4.0

### ISSUE THREAD

**Issue by emazep** [matplotlib/matplotlib#19803]
If I run the following code:

```python
import matplotlib.pyplot as plt
plt.subplots()
plt.suptitle('SUPTITLE TEST')
```

I get:

![image](https://user-images.githubusercontent.com/152575/112739947-f689b700-8f78-11eb-883a-acecd74fc603.png)

which says that the title is placed in `x=0.5, y=0.98` (which, according to the docs, is its default position). However if I do:

```python
plt.suptitle('SUPTITLE TEST', x=0.5, y=0.98)
```

it still says that the text is placed in `x=0.5, y=0.98`, but it is completely disappeared:

![image](https://user-images.githubusercontent.com/152575/112739976-3a7cbc00-8f79-11eb-8f79-d3a9bc85760f.png)

I think because it goes out of the image boundaries, as if I do:

```python
plt.suptitle('SUPTITLE TEST', y=0.93)
```

I get the title vertically cut, instead of obtaining the space to accommodate it:

![image](https://user-images.githubusercontent.com/152575/112740020-99423580-8f79-11eb-86f5-c7cbe8118494.png)

Also, setting `x` to any value but not setting `y`, such as in:

```python
plt.suptitle('SUPTITLE TEST', x=0.5)
```

makes the title disappear as well:

![image](https://user-images.githubusercontent.com/152575/112740040-d4dcff80-8f79-11eb-9b0e-ba586f49dc9d.png)

I can also cut the title horizontally, for example with:

```python
plt.suptitle('SUPTITLE TEST', x=0.1, y=0.92)
```

which produces:

![image](https://user-images.githubusercontent.com/152575/112740132-b62b3880-8f7a-11eb-99b9-08ece0aef052.png)

Basically I cannot position the title anymore, while it was easily doable until 3.3.4.

Thank you.

**Matplotlib version**
  * Operating system: any
  * Matplotlib version: 3.4.0 
  * Matplotlib backend: module://ipykernel.pylab.backend_inline 
  * Python version: 3.9.2

**Comment by jklymak** [comment]
Sorry about this - it is the constrained layout rewrite.  Hopefully an easy fix

**Comment by jklymak** [comment]
BTW< a work around for your manually-placed subtitles is `tlt = plt.suptitle('Boo'); tlt.set_in_layout(True)`.

**Comment by jklymak** [comment]
This should be fixed in 3.4.1.  Sorry for the bother

**Comment by emazep** [comment]
> 
> 
> This should be fixed in 3.4.1. Sorry for the bother

Don't mention it and thank you for your quick fix. This library not only has a nearly infinite power and flexibility, but also a stellar support!

**Comment by jklymak** [comment]
Well, the stellar support at least tries to fix things they gratuitously broke :wink:
